### DISCUSSION TITLE
Improve the description of PII data provided for AFN orders

### DISCUSSION THREAD

**Discussion by boesbo** [amzn/selling-partner-api-models#3275] [General]
The description of the PII data provided for AFN orders should be improved: https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/roles/Roles-in-the-Selling-Partner-API.md

For example Amazon support told me that despite having the "Direct-to-Consumer Shipping (Restricted)" role, for AFN orders, the "name" is not provided:
```
We understand that you require Names of buyers on AFN orders.

AFN orders is Fulfilled by Amazon orders.

PII is by default not provided for orders fulfilled by Amazon in order to protect customer data.

Sellers who need assistance with their FBA orders may contact our Fulfillment by Amazon department using the Contact Us link at the bottom of Seller Central pages. If you have any other questions, please let us know.
```

This is not documented, and I don't know today if the "name" field of AFN orders comes with the "Tax Invoicing (Restricted)" role, for tax reasons.

We should clarify once and for all.

**Comment by NetRushJacob** [comment]
Looking at a response for an AFN fulfillment channel, the Orders API returns a BuyerInfo node that contains BuyerEmail, BuyerName populated. BuyerCountry, BuyerTaxInfo and PurchaseOrderNumber were all null in the random sampling of 5 orders.

Note that the Data Protection Policy is pretty explicit when it comes to this information.
https://sellercentral.amazon.com/mws/static/policy?documentType=DPP&locale=en_US

**Reply by boesbo** [reply]
I opened an Amazon case in the past few days, and they told me that for AFN orders, despite having PII authorization, it is normal for the "BuyerName" field to be null. But now you let me know that this is not the case and that the name field is populated.I can not understand :(
It seems like a never ending story, I don't understand why assistance has to say wrong things and waste time.

Amazon response:
```
My name is ****, from the Developer Support team.

We understand that you require Names of buyers on AFN orders.

AFN orders is Fulfilled by Amazon orders.

PII is by default not provided for orders fulfilled by Amazon in order to protect customer data.

Sellers who need assistance with their FBA orders may contact our Fulfillment by Amazon department using the Contact Us link at the bottom of Seller Central pages. If you have any other questions, please let us know.
```

How is it possible that you see the "BuyerName"?

**Reply by NetRushJacob** [reply]
Uncertain on what would cause us to get that information while not allowing others. 
Are you submitting the request with the Restricted Data Token (RDT)? 

Documentation and understanding could certainly be clarified by Amazon as support seems to be speaking differently than reality.

**Reply by boesbo** [reply]
I also did a test with a different developer account associated with my Amazon Seller account, with MWS access, same problem for AFN orders, the BuyerName field is null, instead for MFN orders the field is correctly populated.
As you say it is good that the documentation points to maximum clarity, because unfortunately the technical assistance is inadequate, and without the support of the documentation it is impossible to solve any problem, because they work aiming to close the case as quickly as possible, they do not think about solving it.
